x-pack/plugins/security/server/authentication/api_keys/api_keys.ts

```diff
@@ -23,16 +26,25 @@ export interface ConstructorOptions {
   license: SecurityLicense;
 }
 
-/**
- * Represents the params for creating an API key
- */
-export interface CreateAPIKeyParams {
+interface BaseCreateAPIKeyParams {
   name: string;
-  role_descriptors: Record<string, any>;
   expiration?: string;
   metadata?: Record<string, any>;
+  role_descriptors: Record<string, any>;
+  role_descriptors_with_kibana_privileges: {
+    role_descriptors: Record<string, any>;
+    kibana_privileges: KibanaPrivileges;
```

Comment:
```suggestion
    role_descriptors: Record<string, any>;
    kibana_privileges: KibanaPrivileges;
```

Sorry if this wasn't clear, but we were looking for Kibana roles here, not a combination of ES roles & Kibana privileges. I would expect this interface to look _something_ like:

```ts
kibana_role_descriptors: Record<string, Pick<Role, 'elasticsearch' | 'kibana'>
```

Where `Role` is defined as:
https://github.com/elastic/kibana/blob/6ecffcc57c9a5b0328257cd25b737140a8471dda/x-pack/plugins/security/common/model/role.ts#L31-L47